python/ray/data/tests/test_dataset.py

```diff
@@ -3556,6 +3556,14 @@ def test_sort_simple(ray_start_regular_shared):
     assert ds.count() == 0
 
 
+def test_column_name_type_check(ray_start_regular_shared):
+    df = pd.DataFrame({"1": np.random.rand(10), "2": np.random.rand(10)})
+    ray.data.from_pandas(df)
+    df = pd.DataFrame({1: np.random.rand(10), 2: np.random.rand(10)})
+    with pytest.raises(ValueError):
+        ray.data.from_pandas(df)
```

Comment:
Could we also add unit tests covering the pandas schema string (e.g., str(df) output for a pandas schema dataset)?